lib/opencensus/trace/span_context.rb

```diff
@@ -123,12 +123,19 @@ def to_trace_context_header
       # However, you are responsible for finishing the span yourself.
       #
       # @param [String] name Name of the span
+      # @param [Sampler] sampler Span-scoped sampler. If not provided,
+      #     defaults to the trace configuration's default sampler.
+      #
       # @return [SpanBuilder] A SpanBuilder object that you can use to
       #     set span attributes and create children.
       #
-      def start_span name, skip_frames: 0
+      def start_span name, skip_frames: 0, sampler: nil
         child_context = create_child
-        span = SpanBuilder.new child_context, skip_frames: skip_frames + 1
+        sampler ||= OpenCensus::Trace::Samplers::DEFAULT
```

Comment:
With the current default (probability at 10%), it's unlikely the user will see any in-depth trees. Should we switch the local default sampler to `AlwaysSample`?

Reply:
The spec for the Probability sampler says that if the parent has been sampled, then the child shall also be sampled. (https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/Sampling.md). I went ahead and implemented that.

FWIW the default probability for the Java sampler is 10^-4...

Reply:
It seems like if a user does specify a local sampler we'd override the sampling decision even if it's the same type of sampler. I would expect that if no local sampler is provided, that we'd inherit the decision from our parent.

Reply:
We can clarify the sampling decision inheriting later.